### PR TITLE
Fix for storing screenshots on Android M

### PR DESCRIPTION
**Pull request by dannyroa**
No need for storage permissions; use context.getExternalFilesDir instead to store screenshots.

### CODE REVIEW

File: spoon-client/src/main/java/com/squareup/spoon/Spoon.java [hunk 1]
```diff
@@ -206,7 +205,7 @@ private static File filesDirectory(Context context, String directoryType, String
     File directory;
     if (Build.VERSION.SDK_INT >= 21) {
       // Use external storage.
-      directory = new File(getExternalStorageDirectory(), "app_" + directoryType);
+      directory = new File(context.getExternalFilesDir(null), "app_" + directoryType);
```
Comment (edenman) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: does this still work with `SpoonDeviceRunner`'s expectation of where the screenshots will be?  I don't have a Spoon rig set up at the moment or I'd just install locally and confirm.

Comment (dannyroa) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: Actually, I missed that. It won't work.

The new path looks like this:

/storage/emulated/0/Android/data/com.joelapenna.foursquared/files/app_spoon-screenshots/com.joelapenna.foursquared.TopPicksTest/goToVenue/1453506199826_goToVenue_end.png

I'm not sure if the format `/storage/emulated/0/Android/data/package_name/app_spoon-screenshots/test_class_name/filename` will be consistent.

Comment (JakeWharton) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: We can switch to parsing the logs for files to pull. I looked at this a few years ago and it didn't seem hard, I was just lazy.

File: spoon-client/src/main/java/com/squareup/spoon/Spoon.java [hunk 2]
```diff
@@ -206,7 +205,7 @@ private static File filesDirectory(Context context, String directoryType, String
     File directory;
     if (Build.VERSION.SDK_INT >= 21) {
```
Comment (tasomaniac) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: I think "storage permission necessity" is removed at API level 19.

Comment (dannyroa) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: Do you have reference to confirm this?

Comment (tasomaniac) [inline spoon-client/src/main/java/com/squareup/spoon/Spoon.java]: Here: 

> For example, beginning with Android 4.4 (API level 19), it's no longer necessary for your app to request the WRITE_EXTERNAL_STORAGE permission when your app wants to write to its own application-specific directories on external storage (the directories provided by getExternalFilesDir())

http://developer.android.com/guide/topics/manifest/uses-permission-element.html#maxSdk

**Comment by dannyroa** [comment]
How do I view the checkstyle report?

**Comment by riwnodennyk** [comment]
Nice! Really missing this fix. Any plans to merge this into the master?

**Comment by Egorand** [comment]
There's a test suite in our project that checks that the Storage permission dialogs are displayed properly, and expects the permission to not be granted by default. We're not able to run this test suite with Spoon, since it grants the permission before the run, breaking the precondition. That's a very helpful pull request, hoping it will be merged soon.

**Comment by edenman** [comment]
@dannyroa can confirm, but I'm pretty sure this PR doesn't actually solve the problem

**Comment by dannyroa** [comment]
This is about avoiding the Storage Permissions.

Getting the accurate directory though is not done yet. Plan is to print it on the logs and read from it.

Haven't had time to work on this so I'm closing it.

**Pull request closed without merging**